### ISSUE TITLE
Request: modules

### ISSUE THREAD

**Issue by siraben** [blynn/compiler#2]
Thanks for your work on this, we've been actively working on [a fork](https://github.com/oriansj/blynn-compiler) of this compiler that bootstraps it all the way from a [357 byte binary](https://github.com/oriansj/mescc-tools-seed) on various architectures. One feature that is missing in particular is modules, which would allow separation of code into different files. 

It's non-critical for our work but if added would greatly help, especially since we plan to build a Scheme interpreter on top of this. How complex would it be to add?

**Comment by blynn** [comment]
I've been wanting to add modules for a while. I've been holding off because I wanted to explore a little, and try implementing Agda-style modules for example. But perhaps it's best just to get something simple working first.

My experimental "precisely" compiler supports #include directives, which I use to split code across multiple files. I'll see if it makes sense to document it and make it a bit more official.

**Comment by blynn** [comment]
On closer inspection, it looks like "crossly" was my first compiler to support `#include`. It's just undocumented.

So if you manage to get "crossly" working, you can temporarily use includes to get a poor man's modules.

**Comment by siraben** [comment]
Yeah, we have blynn-compiler compiling all the way to precisely. Thanks!

**Comment by blynn** [comment]
I added modules, though there are many caveats at present. It's based on "methodically", and I hope to work my way back up to "crossly" via modules. https://crypto.stanford.edu/~blynn/compiler/module.html

**Comment by siraben** [comment]
Are features like qualified imports and `hiding` planned?

**Comment by blynn** [comment]
I hope to get around to it some time.

**Comment by blynn** [comment]
I have a rough implementation of qualified imports and hiding, though modules in general are rough around the edges.
